electrumsv/tests/test_wallet.py

```diff
@@ -486,128 +508,346 @@ def test_multisig(self, mock_app_state, tmp_storage) -> None:
-
-    # When a wallet is composed of multiple files, we need to know which to load.
-    wallet_filenames = []
-    if storage_info.kind != StorageKind.DATABASE:
-        wallet_filenames.append(storage_info.filename)
-    if storage_info.kind in (StorageKind.DATABASE, StorageKind.HYBRID):
-        wallet_filenames.append(storage_info.filename + DATABASE_EXT)
+def load_wallet(storage: WalletStorage) -> Wallet | None:
+    try:
+        return Wallet(storage)
+    except FileNotFoundError as e:
+        if sys.version_info[:3] >= (3, 8, 0):
+            msg = "Could not find module 'libusb-1.0.dll' (or one of its dependencies)."
+            if msg in e.args[0]:
+                pytest.xfail("libusb DLL could not be found")
+                return
+        raise e
+    except OSError as e:
+        if sys.version_info[:3] < (3, 8, 0):
+            if "The specified module could not be found" in e.args[1]:
+                pytest.xfail("libusb DLL could not be found")
+                return
+        raise e
+
+
+class ReEncryptionTestState(TypedDict):
+    """This is state that is persisted across all parametrized test runs to give a final report
+    of the test coverage of re-encrypting all types of private encryption data. The idea is that
+    we have created a selection of test wallets that cover all the different types of encrypted
+    data. If any of the `count_` prefixed fields have not been incremented from zero, then
+    test coverage is not complete.
+
+    These are currently up to four categories of encrypted data that must be re-encrypted in both
+    the database and in any corresponding application caches for a successful wallet password
+    update:
+        - password_token in the `WalletData` table
+        - derivation data in the `KeyInstances` table sometimes contains private keys for
+          `ImportedPrivkeyAccount` types
+        - derivation_data in the `MasterKeys` table contains various types of encrypted data
+          depending on the type of wallet or how old it is. These are the possible lookup keys in the
+          json:
+            - "seed"
+            - "passphrase"
+            - "xprv"
+        - encrypted_api_key data in the `Servers` table
+
+    New encrypted data may be added in future versions. Therefore, it is important to write this
+    test in such a way that it:
+        a) Captures any regressions in the re-encryption of migrated old wallets
+        b) Is easy to extend to cover any new encrypted data that is added in future versions
 
-    temp_dir = tempfile.mkdtemp()
-    for _wallet_filename in wallet_filenames:
-        source_wallet_path = os.path.join(TEST_WALLET_PATH, _wallet_filename)
-        wallet_path = os.path.join(temp_dir, _wallet_filename)
-        shutil.copyfile(source_wallet_path, wallet_path)
+    """
+    total_wallets_tested: int
+    failing_wallets: set[str]
+
+    # ---- Encrypted Data Types Requiring Test Coverage For Re-encryption ---- #
```

Comment:
What is this random comment style? Get rid of this and follow a normal style. Use a double hash or something.